src/nipple.js

```diff
@@ -240,9 +253,12 @@ Nipple.prototype.hide = function (cb) {
         },
         self.options.fadeTime
     );
-    if (self.options.restJoystick) {
-        self.setPosition(cb, { x: 0, y: 0 });
-    }
+
+
+    self.setPosition(cb, { 
+        x: self.options.restJoystick.x ?0 :self.instance.frontPosition.x, 
+        y: self.options.restJoystick.y ?0 :self.instance.frontPosition.y 
+    });
```

Comment:
I think it's best to have this logic just once where we do actually use it.
You can remove the other logics that you duplicated.

```suggestion
    if (self.options.restJoystick) {
        const rest = self.options.restJoystick;
        const newPosition = {};
        newPosition.x = rest === true || rest.x ? 0 : self.instance.frontPosition.x;
        newPosition.y = rest === true || rest.y ? 0 : self.instance.frontPosition.y;
        self.setPosition(cb, newPosition);
    }
```

Reply:
refactored it as you request..